project-base/storefront/components/Blocks/Adverts/Adverts.tsx

```diff
@@ -42,11 +42,11 @@ export const Adverts: FC<AdvertsProps> = ({
     const content = !!displayedAdverts?.length && (
         <div className={twJoin(withGapBottom && 'mb-8', withGapTop && 'mt-8', !withWebline && className)}>
             {displayedAdverts.map((advert, index) => {
-                if (advert.__typename === 'AdvertImage') {
-                    if (!shouldBeShown(advert, positionName, currentCategory)) {
-                        return null;
-                    }
+                if (!shouldBeShown(advert, positionName, currentCategory)) {
+                    return null;
+                }
 
+                if (advert.__typename === 'AdvertImage') {
```

Comment:
Even though I am usually more inclined towards doing array iteration in one run, in this case it makes more sense to first filter, and then map, meaning:

```tsx
{displayedAdverts
    .filter(advert => shouldBeShown(advert, positionName, currentCategory))
    .map((advert, index) => {
        if (advert.__typename === 'AdvertImage') 
```

Reply:
good idea, thanks 🙂